[DOCS-6429] Add OP Datadog Archives (#20330)
DataDog/documentation: content/en/observability_pipelines/guide/set_up_datadog_archives_destination.md

### Configuration file

For manual deployments, the [sample pipelines configuration file][7] for Datadog includes a sink for sending logs to Amazon S3 under a Datadog-rehydratable format. 

{{< tabs >}}
{{% tab "Docker" %}}

In the sample pipelines configuration file, replace `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` with the AWS credentials you created earlier.

{{% /tab %}}
{{% tab "AWS EKS" %}}

In the sample pipelines configuration file, replace `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` with the AWS credentials you created earlier.

{{% /tab %}}
{{% tab "APT-based Linux" %}}

In the sample pipelines configuration file, replace `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` with the AWS credentials you created earlier.

{{% /tab %}}
{{% tab "RPM-based Linux" %}}

In the sample pipelines configuration file, replace `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` with the AWS credentials you created earlier.

{{% /tab %}}
{{% tab "Terraform (AWS)" %}}

Replace `${DD_ARCHIVES_BUCKET}` and $`{DD_ARCHIVES_REGION}` parameters based on your S3 configuration.

{{% /tab %}}
{{< /tabs >}}

> <div class="alert alert-warning">If you are routing logs ingested into the OP Worker that are not coming from the Datadog Agent, to an archive with the <code>datadog_archives</code> destination, those logs are not tagged with <a href="https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#reserved-attributes">reserved attributes</a>. This means that you lose Datadog telemetry and the benefits of <a href="https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes">unified service tagging</a>. For example, say your syslogs are sent to <code>datadog_archives</code> and those logs have the status tagged as <code>severity</code> instead of the reserved attribute of <code>status</code> and the host tagged as <code>hostname</code> instead of the reserved attribute <code>hostname</code>. When these logs are rehydrated in Datadog, the <code>status</code> for the logs are all set to <code>info</code> and none of the logs will have a hostname tag.</div>

Reviewer: This one I'm less sure about, if it makes it more confusing, leave as it.

> <div class="alert alert-warning">If you are routing logs ingested into the OP Worker that are not coming from the Datadog Agent, to an archive with the <code>datadog_archives</code> destination, those logs are not tagged with <a href="https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#reserved-attributes">reserved attributes</a>. This means that you lose Datadog telemetry and the benefits of <a href="https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes">unified service tagging</a>. For example, say your syslogs are sent to <code>datadog_archives</code> and those logs have the status tagged as <code>severity</code> instead of the reserved attribute of <code>status</code> and the host tagged as <code>hostname</code> instead of the reserved attribute <code>hostname</code>. When these logs are rehydrated in Datadog, the <code>status</code> for the logs are all set to <code>info</code> and none of the logs will have a hostname tag.</div>

Author: I edited it so it's hopefully clearer.

{{< tabs >}}
{{% tab "AWS S3" %}}

7. In the **Bucket** field, enter the name of the S3 bucket you created earlier.
8. Enter `aws_s3` in the **Service** field.
9. Toggle **AWS S3** to enable those specific configuration options.
10. In the **Storage Class** field, select the storage class in the dropdown menu.
11. Set the other configuration options based on your use case.

{{% /tab %}}
{{% tab "Azure Blob" %}}

7. In the **Bucket** field, enter the name of the S3 bucket you created earlier.
8. Enter `azure_blob` in the **Service** field.
9. Toggle **Azure Blob** to enable those specific configuration options.
10. Enter the Azure Blob Storage Account connection string.
11. Set the other configuration options based on your use case.

{{% /tab %}}
{{% tab "GCP Cloud Storage" %}}

7. In the **Bucket** field, enter the name of the S3 bucket you created earlier.
8. Enter `gcp_cloud_storage` in the **Service** field.
9. Toggle **GCP Cloud Storage** to enable those specific configuration options.
10. Set the configuration options based on your use case.

{{% /tab %}}
{{< /tabs >}}

See [Datadog Archives reference][9] for details on all configuration options.

## Rehydrate your archive

> See [Rehydrating from Archives][10] for instructions on how to rehydrate your archive in Datadog so that you can start analyzing and investigating those logs.

Reviewer: ```suggestion
Rehydrate archived logs to analyze and investigate log events. For more information and instructions, see [Rehydrating from Archives][10].
```
